src/multiLanguageDocumentList.js

```diff
@@ -1,50 +1,185 @@
 import S from '@sanity/desk-tool/structure-builder'
 import { liveDocuments } from '@kaliber/sanity-live-documents'
-import { map, shareReplay } from 'rxjs/operators'
+import { map } from 'rxjs/operators'
 import React from 'react'
 import * as rxjs from 'rxjs'
 import { Title } from './Title'
 import schema from 'part:@sanity/base/schema'
-import {ComposeIcon} from '@sanity/icons'
-
-export function multiLanguageDocumentList({ schemaType, id, titleField, spec = undefined }) {
-  const type = schema.get(schemaType)
-  /** @type {rxjs.Observable<any>} - you can remove this in your spare time */
-  const documents$ = liveDocuments({ schemaType })
-  const liveTranslationGroups = documents$
-    .pipe(
-      map(toTranslationGroups),
-      shareReplay({ bufferSize:1, refCount: false })
+import { usePane, usePaneLayout } from '@sanity/desk-tool/lib/components'
+import { usePaneRouter } from '@sanity/desk-tool'
+import { ListPaneContent } from './machinery/sanity/ListPaneContent'
+import { PaneItem } from './machinery/sanity/PaneItem'
+import { useDeskToolSettings } from './machinery/sanity/useDeskToolSettings'
+
+export function multiLanguageDocumentList(listBuilder) {
+  const serializedListBuilder = listBuilder.serialize()
+
+  const document$ = new rxjs.BehaviorSubject({ _type: serializedListBuilder.schemaTypeName })
+  const component = React.memo(React.forwardRef(MultiLanguageDocumentList))
+  return {
+    ...serializedListBuilder,
+    type: 'component', component,
+    key: serializedListBuilder.id, // override the key to prevent an unmount / mount cycle
+    document$,
+    child: document$.pipe(
+      map(doc =>
+        getDocumentNode({ schemaType: doc._type, documentId: doc._id })
+          .id(doc.translationId || 'unknown')
+      )
     )
-S.defaultInitialValueTemplateItems
-  return liveTranslationGroups.pipe(
-    map(translations =>
-      S.list(spec).id(id)
-        .items(
-          translations.map(({ translations, translationId }) => {
-            const [first] = translations
-
-            const document$ = new rxjs.BehaviorSubject(first)
-
-            const title = (
-              <Title
-                {...{ titleField, document$, translations }}
-                onDocumentClick={doc => document$.next(doc)}
-              />
-            )
-            const child = document$.pipe(
-              map(doc => getDocumentNode({ schemaType, documentId: doc._id }).id(translationId))
-            )
-            /** @type {any} - you can remove this in your spare time */
-            const listItem = { id: translationId, child, title, type: 'listItem' }
-            return listItem
-          })
-        )
+    // Ik denk dat we in de actiebalk vlaggetjes moeten plaatsen waarmee je kunt switchen voor de display van de taal (tenzij een document niet beschikbaar is in die taal)
```

Comment:
Wat is precies de actiebalk? Bedoel je de balk helemaal bovenin? Volgensmij kunnen we daar alleen acties in hacken, omdat er geen officieel haakje is. Via het logo of via een wrapper om de 'tools' (desktool, vision, etc.) heen.

Reply:
Bij een pane kun je menu items opgeven. Dit kunnen intents en actions zijn. Die bedoel ik :-D